### PR TITLE
chore: migrate to new versions of prosemirror packages

### PR DESCRIPTION
**Pull request by bdbch**
Since Prosemirror switched to Typescript in their new version Tiptap was broken on a fresh install because of a non-fixed version requiring a workaround via overrides. See this issue: #2836

In this PR I'm working on a fix for that to migrate to the new prosemirror versions by default.

### CODE REVIEW

**Comment by anton-liubushkin** [comment]
🔥

File: packages/extension-code-block-lowlight/package.json [hunk 1]
```diff
@@ -27,13 +27,14 @@
   "dependencies": {
     "@tiptap/extension-code-block": "^2.0.0-beta.40",
     "@types/lowlight": "^0.0.3",
-    "prosemirror-model": "^1.16.1",
-    "prosemirror-state": "^1.3.4",
-    "prosemirror-view": "^1.23.6"
+    "lowlight": "^1.20.0",
```
Comment (enriquecastl) [inline packages/extension-code-block-lowlight/package.json:30]: Hi @bdbch! 

I’m trying to [upgrade tiptap packages](https://gitlab.com/gitlab-org/gitlab/-/merge_requests/88851#note_999312652) in our application, and I noticed that in the latest released version of `code-block-lowlight` declares the `lowlight` package as a dependency **and** a peerDependency: https://unpkg.com/browse/@tiptap/extension-code-block-lowlight@2.0.0-beta.72/package.json

This is not the case in the master branch: https://github.com/ueberdosis/tiptap/blob/main/packages/extension-code-block-lowlight/package.json#L25. 

What I find confusing is that part of the changes that I submitted in the [PR that moves lowlight to a peerDependency](https://github.com/ueberdosis/tiptap/pull/2625/files) are also included in the latest released version, which differs from what exists in the master branch. 

Why is the package.json file different in the master branch and the released package?

**Comment by rootlinux2** [comment]
Any advance here?

**Comment by bdbch** [comment]
This is already merged @rootlinux2 the problem you described is coming from the prosemirror-tables package. I'll fork it and work on it to make it work with cjs / require syntax.

**Comment by rootlinux2** [comment]
@bdbch, @amursky 
I updated tip-tap to the latest versions

image

but I'm still facing this error!

```
Error: Must use import to load ES Module: /home/rootlinux2/Documents/Front10/apexedge-widgets/node_modules/prosemirror-tables-contently/dist/index.js
require() of ES modules is not supported.
require() of /home/rootlinux2/Documents/Front10/apexedge-widgets/node_modules/prosemirror-tables-contently/dist/index.js from /home/rootlinux2/Documents/Front10/apexedge-widgets/node_modules/@tiptap/extension-table/dist/tiptap-extension-table.cjs.js is an ES module file as it is a .js file whose nearest parent package.json contains "type": "module" which defines all .js files in that package scope as ES modules.
Instead rename index.js to end in .cjs, change the requiring code to use import(), or remove "type": "module" from /home/rootlinux2/Documents/Front10/apexedge-widgets/node_modules/prosemirror-tables-contently/package.json.
```

**Comment by bdbch** [comment]
Hey @rootlinux2 we'll publish a new release ASAP. I'll let you know.

**Pull request merged**